### ISSUE TITLE
broken intellisense is making zed unusable 

### ISSUE THREAD

**Issue by pchalasani** [zed-industries/zed#22266]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

The various AI features are great but the very basic code completion seems broken. 
To try to get this working, I commented out the entire "languages" section in my `settings.json`, and uninstalled all relevant extensions, but can't seem to get basic path-completions to work.

E.g. if I do `from blah.` I expect to see (good) auto-complete suggestions, but nothing shows up.
Sometimes I've seen it show a drop-down of choices, but sometimes (more often) I don't, so I've no idea what's going on, nor is there anything helpful in discord or docs. 

### Zed Version and System Specs

Zed: v0.166.1 (Zed)
OS: macOS 15.0.0
Memory: 64 GiB
Architecture: aarch64

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary>

```

```
</details>

**Comment by jansol** [comment]
What languages specifically? And what do you mean by "path-completions"?

**Comment by pchalasani** [comment]
> What languages specifically? And what do you mean by "path-completions"?

Python.

By path-completions I just meant import statements like `from blah.` where I pause at the "dot" and expect suggestions/completions.

**Comment by pchalasani** [comment]
Also maybe "intellisense" is not the exact technical term, but what I mean is suggestions where it shows the best/likeliest choice of class/function etc. There may be two "competing" completions - copilot and the ones from the LSP, but I don't know where to specify which to prefer (or I don't fully understand the interplay).

**Comment by EnnFour** [comment]
I'm running into similar issues with unusable intellisense -- pretty much everything suggested is an import from a different module. Here's an example of me attempting to annotate one of my methods with the @abstractmethod decorator. I've already imported abstractmethod:
![image](https://github.com/user-attachments/assets/5a3568db-c962-4602-88ad-6785ff5deee8)

I'd love some toggleable settings which
- enable/disable suggestions which require an import
- enable/disable suggestions within a `from x import y` block (though I'd also be happy with the ability to selectively enable autocompletions for the x component or the y component)

**Comment by pchalasani** [comment]
I'm thinking this is probably mainly a docs issue and not a bug -- there needs to be a definitive document somewhere that shows, "if you want to have x,y,z capabilities, this is what your settings.json should contain", etc. Currently I'm finding partial and inconsistent success by hunting down these settings in various gh issues and even reddit. The latest incarnation of my hundreds of iterations of `settings.json` that *seems* to work for me (at least until it breaks on another update) is below. 
The module auto-complete is sort of working but nowhere near as good as JetBrains/IntelliJ -- 

e.g. when I do `from blah.foo import MyModule`, now I do get good suggestions when I enter ".", but as soon as I type the first letter "M", I expect to see suggestions immediately (which I do in JetBrains), but I don't; I have to type the _second_ letter "y" and only then I see suggestions.

```json
 "languages": {
      "Python": {
        "language_servers": ["pyright", "ruff", "!python-refactoring", "!pylsp"],
        "enable_language_server": true,
        "format_on_save": "on",
        "formatter": [
          {
            "code_actions": {
              "source.organizeImports.ruff": true,
              "source.fixAll.ruff": true
            }
          },
          {
            "language_server": {
              "name": "ruff"
            }
          }
        ]
      }
    },
    
  "lsp": {
      "pyright": {
        "settings": {
          "python.analysis": {
            "ignore": ["*"],
            "diagnosticMode": "workspace",
            "typeCheckingMode": "strict"
          },
          "python": {
            "pythonPath": ".venv/bin/python"
          }
        }
      },
      "ruff": {
            "initialization_options": {
              "settings": {
                // Ruff server settings goes here
                "lineLength": 89,
                "lint": {
                  "extendSelect": ["I", "F", "E"]
                }
              }
            }      
      },
      "pylsp": {
            "settings": {
              "plugins": {
                "mypy": {
                  "enabled": true
                },
                "pycodestyle": {
                  "enabled": false
                },
                "flake8": {
                  "enabled": false
                },
                "rope": {
                  "enabled": false
                },
                "pyflakes": {
                  "enabled": false
                },
                "mccabe": {
                  "enabled": false
                }
              }
            }
      },      
    },
```

**Comment by mstephenson6** [comment]
I got here by googling "zed python broken", which was exactly my experience after trying to edit a very simple python script, one of those random utils in a large repo of another language entirely. The stuff that python is just really good at.

I looked at all the stuff above and went "nope", re-installed VSCode, and now I'm editing Python there again, wondering if I should just come back to it wholesale after this surprise. I've been using Zed Preview for about six months in many languages, Python is the first complete miss.

I do not understand why Python gets listed as a supported language, without basic autocomplete for its standard library.

**Comment by pchalasani** [comment]
> now I'm editing Python there again, wondering if I should just come back to it wholesale after this surprise.

Yes I keep getting squiggles (due to ruff) in the zed editor, I know I can turn those off somehow but the process isn't smooth. And auto-complete is either flaky or suggests bad things, not as good as JetBrains. 

My (almost comically complex) workflow is to have the same kb shortcuts in VSCode, Zed and JetBrains and use each for what it's good at:
- JetBrains/Pycharm for a very pleasant basic editing experience with excellent autocomplete and jump to def etc
- VSCode for reviewing Pull reqs and using roo-cline extension for multi-step file create + edit stuff
- Zed for being able to just modify code inline by hitting Ctl+Enter in the code, and interact with the LLM in the Assistant Pane, and also Ctl+Enter in the terminal view for CLI help.

**Comment by eddiebergman** [comment]
Just in defense of `zed`, I've gone down the rabbit hole of trying to set up a good environment to solve all of the aforementioned problems with Neovim and it's not entirely zeds fault, but it's a hard problem because no language server out there has solved it. Improving it would take a lot of effot.

tldr;
a) Intellisense -> LSP -> `pyright`
b) Python just funky
c) Microsoft don't want to improve `pyright`
d) See this: https://github.com/microsoft/pyright/discussions/7099
---

Just for reference for people coming here and do no know what _intellisense_ really is, it's almost always provided by a language-sever speaking the [LSP](https://microsoft.github.io/language-server-protocol/), in the case, `pyright` for python. If you type `:language` in zed and choose the option to open the logs for `pyright`, you can see that it will index (by default) your entire python environment. There's recorded issues of this, where essentially the amount of possible ways to import the same symbols is **large**, as people rarely use the `__all__` property of a python module to publicly advertise what to export. For example:

```python
# my_module.py
from dataclasses import dataclass # python builtin
from foo import Blah

@dataclass
class MyClass:
    blah: Blah

# other_module.py
from my_module import <ask-for-completion>
```

In this case, even though the intent of `my_module.py` is really just define a class `pyright` sees 3 different possible imports.
```python
from my_module import dataclass
from my_module import MyClass
from my_module import Blah
```

Now suppose you have a host of libraries installed, and all of them import some builtin, or they all use some lower-level library and import `Flooper` from it. When you want to import `Flooper`, there can sometimes be up to hundreds of possible files you could import it from, how do you know which one is the canonical one, automatically?

Last part, this is what sets `pycharm` and `vscode` appart, they have put in hundreds of engineering hours into trying to do "the right thing" in a language such as python, and consequently, that's why Microsoft use `pylance` in `vscode`, which is closed source and conveniently solves all these issues to make vscode great, and `pycharm` who pride themselves and a few select, well supported languages.

I don't think zed have that kind of manpower to fix these issues, they just get a bunch of messages from the `pyright` language server and that's all they can reasonably show without going down rabbit hole of "just make it do the right thing", which is nigh-impossible with Python.

**Comment by pjv** [comment]
@eddiebergman have you looked into [basedpyright](https://docs.basedpyright.com/latest/)? There’s an installable 3rd-party [extension](https://github.com/m1guer/basedpyright-zed) for zed.

**Comment by eddiebergman** [comment]
> [@eddiebergman](https://github.com/eddiebergman) have you looked into [basedpyright](https://docs.basedpyright.com/latest/)? There’s an installable 3rd-party [extension](https://github.com/m1guer/basedpyright-zed) for zed.

Have not seen this before but it seems based :) Will have to give it a try!

**Comment by jacobduba** [comment]
Can confirm it's still broken, I'm running into the same issue. Sorry to the Zed developers that I don't have time right now to debug more.

**Comment by JosephTLyons** [comment]
We'll take a look at this, sorry we dropped the ball here.
